### PR TITLE
Abhinav -- Firmware Onboarding Modules (101-103)

### PR DESCRIPTION
**Pull request by Abhinav-Bala**

### CODE REVIEW

File: projects/gpio_task/src/main.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  gpio_it_init();
+
+  gpio_init_pin(&led1, GPIO_OUTPUT_PUSH_PULL, GPIO_STATE_HIGH);
+  gpio_init_pin(&led2, GPIO_OUTPUT_PUSH_PULL, GPIO_STATE_HIGH);
+  gpio_init_pin(&led3, GPIO_OUTPUT_PUSH_PULL, GPIO_STATE_HIGH);
+  gpio_init_pin(&led4, GPIO_OUTPUT_PUSH_PULL, GPIO_STATE_HIGH);
+  gpio_init_pin(&button, GPIO_INPUT_PULL_DOWN, GPIO_STATE_LOW);
+
+  tasks_init();
+  tasks_init_task(gpio_led, TASK_PRIORITY(2), NULL);
```
Comment (Abhinav-Bala) [inline projects/gpio_task/src/main.c:71]: Is there any specific order in which you have to call these functions?

File: projects/gpio_task/src/main.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+#define BTN_INT_EVENT 5
+
+static InterruptSettings settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+  .edge = INTERRUPT_EDGE_RISING,
+};
+
+static GpioAddress led1 = {
```
Comment (Bafran) [inline projects/gpio_task/src/main.c:16]: These could be an array

File: projects/gpio_task/src/main.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+uint32_t notif;
+
+TASK(gpio_led, TASK_STACK_512) {
+  LOG_DEBUG("Initializing GPIO_LED Task!\n");
+
+  while (true) {
+    notify_get(&notif);
+    if (notify_check_event(&notif, BTN_INT_EVENT)) {
+      gpio_toggle_state(&led1);
```
Comment (Bafran) [inline projects/gpio_task/src/main.c:49]: If the initializations went in an array, a for loop here could be an improvement

File: projects/gpio_task/src/main.c (hunk 1 above)
Comment (Bafran) [reply projects/gpio_task/src/main.c:71]: Not necessarily, we just typically do library inits -> task inits -> tasks start

**Review by Bafran — COMMENTED** [review]
Pretty minor comments, looks really good!

**Pull request closed without merging**